### PR TITLE
Change timeout logs/tags

### PR DESCRIPTION
**Pull request by pavolloffay**
Related to https://github.com/opentracing-contrib/java-web-servlet-filter/pull/37#pullrequestreview-128344496

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

**Review by objectiser — APPROVED** [review]
Not reporting timeout as error seems to be consistent with spec, as didn't refer to it in the same sentence as errors. If it changes in the future, then it can be changed back.

File: opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/TracingFilterTest.java [hunk 1]
```diff
@@ -243,18 +243,16 @@ public void testAsyncTimeout() throws IOException {
 
         MockSpan mockSpan = mockSpans.get(0);
         Assert.assertEquals("GET", mockSpan.operationName());
-        Assert.assertEquals(6, mockSpan.tags().size());
+        Assert.assertEquals(5, mockSpan.tags().size());
```
Comment (jpkrohling) [inline opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/TracingFilterTest.java:246]: Why did it change? Is it because of the removal of the error flag?

File: opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/TracingFilterTest.java [hunk 2]
```diff
@@ -243,18 +243,16 @@ public void testAsyncTimeout() throws IOException {
...
         Assert.assertEquals(localRequestUrl("/asyncTimeout"), mockSpan.tags().get(Tags.HTTP_URL.getKey()));
         Assert.assertEquals(500, mockSpan.tags().get(Tags.HTTP_STATUS.getKey()));
         Assert.assertEquals("java-web-servlet", mockSpan.tags().get(Tags.COMPONENT.getKey()));
 
         Assert.assertEquals(1, mockSpan.logEntries().size());
-        Assert.assertEquals(3, mockSpan.logEntries().get(0).fields().size());
-        Assert.assertEquals(Tags.ERROR.getKey(), mockSpan.logEntries().get(0).fields().get("event"));
-        Assert.assertEquals("timeout",
-            mockSpan.logEntries().get(0).fields().get("message"));
+        Assert.assertEquals(2, mockSpan.logEntries().get(0).fields().size());
```
Comment (jpkrohling) [inline opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/TracingFilterTest.java:254]: Same as above: did it change because you removed the error flag?

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java [hunk 3]
```diff
@@ -104,10 +104,8 @@ public void onError(HttpServletRequest httpServletRequest, HttpServletResponse h
         @Override
         public void onTimeout(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
                               long timeout, Span span) {
-            Tags.ERROR.set(span, Boolean.TRUE);
```
Comment (jpkrohling) [inline opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java:107]: Why this change? I would say that a timeout is an error condition, no?

Comment (jpkrohling) [reply opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java:107]: I just noticed @objectiser's comment and it makes sense.

**Review by jpkrohling — APPROVED** [review]

File: opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/TracingFilterTest.java (hunk 1 above)
Comment (pavolloffay) [reply opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/TracingFilterTest.java:246]: The change is subject of this PR :)

**Comment by pavolloffay** [comment]
I am adding a pointer to the spec section regarding timeout https://github.com/opentracing/specification/blob/master/semantic_conventions.md#log-fields-table in the event there is "timed out"
